Core/src/main/java/org/gusdb/fgputil/DelimitedDataParser.java

```diff
@@ -42,19 +43,13 @@ public Optional<Integer> indexOfColumn(String columnName) {
 
   public String[] parseLineToArray(String line) {
     String[] values = line.split(_delimiterRegex, -1);
-    if (_enforceStrictColumns && values.length != _columnNames.length) {
-      throw new RuntimeException("Expected " + _columnNames.length + " columns but parsed " + values.length + " in line:\n" + line);
+    if (_enforceStrictColumns && values.length != _lineBuilder.size()) {
+      throw new RuntimeException("Expected " + _lineBuilder.size() + " columns but parsed " + values.length + " in line:\n" + line);
     }
     return values;
   }
 
-  public LinkedHashMap<String,String> parseLine(String line) {
-    String[] values = parseLineToArray(line);
-    LinkedHashMap<String,String> map = new LinkedHashMap<>();
-    int numToParse = Math.min(values.length, _columnNames.length);
-    for (int i = 0; i < numToParse; i++) {
-      map.put(_columnNames[i], values[i]);
-    }
-    return map;
+  public Map<String,String> parseLine(String line) {
+    return _lineBuilder.build().putAll(parseLineToArray(line));
```

Comment:
I think we can do better than `String.split` performance-wise since we know the length of the output, though it'd be a little annoying to write our own thign. Under the hood they're using an ArrayList which we don't need knowing the size.